### ISSUE TITLE
`jj init --git-repo=.` creates divergent branches if local and remote diverge

### ISSUE THREAD

**Issue by necauqua** [martinvonz/jj#1862]
## Steps to Reproduce the Problem
```bash
git init --bare remote
git init local
cd local
git commit --allow-empty -m 'A'
git commit --allow-empty -m 'B'
git branch tmp
git reset --hard HEAD^
git commit --allow-empty -m 'C'
git remote add origin localhost:$PWD/../remote
git push -u origin
git reset --hard tmp
git branch -d tmp

git log --all --decorate --oneline --graph

jj init --git-repo=.

jj log -r 'all()'
```

## Expected Behavior
My expectation was that the branch `main` would only point to `main@git`

## Actual Behavior
Branch main diverges, pointing to both `main@git` and `main@origin`

## Specifications
jj 0.7.0-152a0683e25c16f0ef57ee20a723755748b696e3

**Comment by ilyagr** [comment]
I'm beginning to think that with our current branch model, what jj does here might be, er, a feature (sorry!). We're talking about `jj init --git-repo .` resulting in something like

```
main@remote     main@git
   main??        main??
    A            B
     \____   ___/
          \ /
           C
```

The problem is that jj has no way of knowing whether the commit A is the local user's work that they mean to abandon or somebody else's work that happened on the remote and shouldn't be lost. Imagine getting to this because a user did `git fetch; jj git init --git-repo .`.

If instead of creating a conflict, jj set `main` definitively at revision `B`, a subsequent `jj git push` would succeed destroying revision `A` on the remote. Again, `A` may have been somebody else's commit. So, it might be reasonable to ask the user to specify which version of `main` they want (B? rebase B on top of A?).

There's also an interesting discussion around https://discord.com/channels/968932220549103686/969829516539228222/1128850385256321085 about how this relates to #1734 and how we might want to change the branching model to allow jj to be more intelligent here.

**Comment by necauqua** [comment]
Yeah as I've said in the initial discord comment, when making the repro I kind of understood that from jjs point of view this does actually make sense, even if slightly unintuitive

So I guess this depends on the branching model change, if it's left unchanged this behavior does make sense - although we may want to document that.

**Comment by yuja** [comment]
I lean towards considering this is a bug.

Under the current branching model, all remote branches are considered to be kept in sync with the local counterparts. So it's probably better for `jj init --git-repo .` to take all remote branches as past state:

1. import git local branches as jj local branches
2. import git remote branches as jj remote branches
3. DO NOT ~~import jj remote branches to local branches~~

If we strictly follow this, missing git local branches become deleted in jj world. This would be too confusing, and the step (3) will need to be adjsted:

1. import git local branches as jj local branches
2. import git remote branches as jj remote branches
3. import jj remote branches to local branches if local_target is none

If we implement some variant of #1734, maybe we can remove the step (3) and leave missing branches as non-tracking:

1. import git local branches as jj local branches
2. import git remote branches as jj remote branches, flag as tracking if local counterpart exists

(For non-tracking->tracking transition in #1734, I agree with https://github.com/martinvonz/jj/issues/1862#issuecomment-1633393966.)

**Comment by yuja** [comment]
> 1. import git local branches as jj local branches
> 2. import git remote branches as jj remote branches
> 3. import jj remote branches to local branches if local_target is none

Wait. `git fetch`/`git pull` often leaves local branches behind remotes. There's no clue that local branch is newer than the remote one if these branches are diverged. So the current behavior makes sense.

**Comment by chriskrycho** [comment]
Forgive me, because it has been a long time, but: what does `hg` do with named branches and bookmarks respectively in similar situations (i.e. pulling from a remote)?

**Comment by yuja** [comment]
In hg, branch names are stored in each commit metadata. Bookmarks are close to git refs, but there are no remote-tracking stuff. iirc conflict resolution is manual step (something like divergent bookmark is renamed to `name@remote`, and user needs to rename/remove it.) The UX was bad.
